pandas/tests/io/formats/test_console.py

```diff
@@ -0,0 +1,71 @@
+import pytest
+
+from pandas.io.formats.console import detect_console_encoding
+
+
+class MockEncoding(object):  # TODO(py27): replace with mock
+    """
+    Used to add a side effect when accessing the 'encoding' property. If the
+    side effect is a str in nature, the value will be returned. Otherwise, the
+    side effect should be an exception that will be raised.
+    """
+    def __init__(self, encoding):
+        super(MockEncoding, self).__init__()
+        self.val = encoding
+
+    @property
+    def encoding(self):
+        return raise_or_return(self.val)
+
+
+def raise_or_return(val):
```

Comment:
Can this be moved as an instance method of `MockEncoding`?

Reply:
I'm also using it down below to test when defaulting to system default -- it covers the same goal of throwing an exception like above here. I could make it a static method of MockEncoding, but I thought using that object for both cases wouldn't make sense since the encoding property is only being consumed from accesses by stdout or stdin. What do you think?

Reply:
@WillAyd did you see this?

Reply:
Let's make this a class method just so all of the relevant constructs stay together